sample_consensus/include/pcl/sample_consensus/impl/sac_model_circle.hpp

```diff
@@ -105,6 +105,26 @@ pcl::SampleConsensusModelCircle2D<PointT>::computeModelCoefficients (const Indic
   return (true);
 }
 
+#ifdef __AVX__
+// This function computes the squared distances (i.e. the distances without the square root) of 8 points to the center of the circle
+template <typename PointT> inline __m256 pcl::SampleConsensusModelCircle2D<PointT>::sqr_dist8 (const std::size_t i, const __m256 a_vec, const __m256 b_vec) const
+{
+  const __m256 tmp1 = _mm256_sub_ps (_mm256_set_ps (input_->points[(*indices_)[i  ]].x, input_->points[(*indices_)[i+1]].x, input_->points[(*indices_)[i+2]].x, input_->points[(*indices_)[i+3]].x, input_->points[(*indices_)[i+4]].x, input_->points[(*indices_)[i+5]].x, input_->points[(*indices_)[i+6]].x, input_->points[(*indices_)[i+7]].x), a_vec);
+  const __m256 tmp2 = _mm256_sub_ps (_mm256_set_ps (input_->points[(*indices_)[i  ]].y, input_->points[(*indices_)[i+1]].y, input_->points[(*indices_)[i+2]].y, input_->points[(*indices_)[i+3]].y, input_->points[(*indices_)[i+4]].y, input_->points[(*indices_)[i+5]].y, input_->points[(*indices_)[i+6]].y, input_->points[(*indices_)[i+7]].y), b_vec);
```

Comment:
Please remove direct access to `points` since we're planning on some internal changes which will affect visibility of points in the future

Reply:
Done (at least in the parts where this PR makes changes)